src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2701,8 +2701,22 @@ BuildRoutesForInsert(Query *query, DeferredErrorMessage **planningError)
 	{
 		InsertValues *insertValues = (InsertValues *) lfirst(insertValuesCell);
 		List *prunedShardIntervalList = NIL;
-		Expr *partitionValueExpr = (Expr *) strip_implicit_coercions(
-			(Node *) insertValues->partitionValueExpr);
+		Node *partitionValueExpr = (Node *) insertValues->partitionValueExpr;
+		partitionValueExpr = strip_implicit_coercions(partitionValueExpr);
```

Comment:
I'm wondering: Is strip_implicit_coercions still necessary if we call eval_const_expressions?

Reply:
This indeed seems like it's not needed anymore. I've removed it now. Tests still pass without it, now that we have `eval_const_expressions`. Lot's of tests start failing when we remove both though, so we do have tests for the effects of `strip_implicit_coercions`. Without both we get lot's of failures with this error:
```

Reply:
Looking at the code for `strip_implicit_coercions` it also removes `CoerceViaIO`, but `eval_const_expressions` replaces it with two function calls. So I think it might be safer to still keep it and call `strip_implicit_coercions` first. I'll take a look on monday if I can create a test that would fail without having it.